### PR TITLE
ari-client: JSDoc annotations added.

### PR DESCRIPTION
**Pull request by dioris-moreno**
Please fill in this template.

- [X] Use a meaningful title for the pull request. Include the name of the package modified.
- [X] Test the change in your own code. (Compile and run.)
- [X] Add or edit tests to reflect the change. (Run with `npm test`.)
- [X] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [X] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [X] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [X] Provide a URL to documentation or source code which provides context for the suggested changes.
- [X] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.
- [X] Include [tests for your changes](https://github.com/DefinitelyTyped/DefinitelyTyped#testing)
- [X] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`. If for reason the any rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]` and not for whole package so that the need for disabling can be reviewed.

### CODE REVIEW

File: types/ari-client/index.d.ts [hunk 1]
```diff
@@ -5,41 +5,164 @@
...
 /// <reference types="node" />
 
+/**
+ *  Create an instance of Client using the provided connection options and call
+ *  the provided callback once the API has been attached to the Client.
+ *
+ *  @param baseUrl - The URL to the ARI instance.
+ *  @param user - The username for the ARI instance.
+ *  @param pass - The password for the ARI instance.
+ *  @param callback - (optional) The callback to be called upon connection.
```
Comment (peterblazejewicz) [inline types/ari-client/index.d.ts:15]:
```suggestion
 *  @param [callback] - The callback to be called upon connection.
```

File: types/ari-client/index.d.ts [hunk 2]
```diff
@@ -5,41 +5,164 @@
...
+}
+
 export interface Client extends Resource {
-    /* Start and Stop Method Overloads */
+    /**
+     *  Creates the WebSocket connection, subscribing to the given apps.
+     *
+     *  @param apps - Name or array of names of the applications to be started.
+     *  @param subscribeAll - Subscribe to all Asterisk events (true/false).
+     *  @param callback - (optional) The callback to be called after applications have started.
```
Comment (peterblazejewicz) [inline types/ari-client/index.d.ts:36]:
```suggestion
     *  @param [callback] - The callback to be called after applications have started.
```

File: types/ari-client/index.d.ts [hunk 3]
```diff
@@ -5,41 +5,164 @@
...
+     */
     start(apps: string | string[], subscribeAll: boolean, callback?: (err: Error, ...args: any[]) => void): void;
+
+    /**
+     *  Creates the WebSocket connection, subscribing to the given apps.
+     *
+     *  @param apps - Name or array of names of the applications to be started.
+     *  @param callback - (optional) The callback to be called after applications have started.
```
Comment (peterblazejewicz) [inline types/ari-client/index.d.ts:44]:
```suggestion
     *  @param [callback] - The callback to be called after applications have started.
```

File: types/ari-client/index.d.ts [hunk 4]
```diff
@@ -142,16 +265,48 @@ export type AnyEventType =
     | ChannelConnectedLineEventType;
 /* Event Classes */
 export interface Events {
-    /* Methods */
+    /**
+     * WebSocket connection for events.
+     *
+     * @param params.app - Applications to subscribe to.
+     * @param params.subscribeAll - (optional) Subscribe to all Asterisk events. If provided, the applications listed will be subscribed to all events, effectively disabling the application
```
Comment (peterblazejewicz) [inline types/ari-client/index.d.ts:272]:
```suggestion
     * @param [params.subscribeAll] - Subscribe to all Asterisk events. If provided, the applications listed will be subscribed to all events, effectively disabling the application
```

File: types/ari-client/index.d.ts [hunk 5]
```diff
@@ -142,16 +265,48 @@ export type AnyEventType =
...
     eventWebsocket(
         params: { app: string | string[]; subscribeAll?: boolean },
         callback: (err: Error, message: Message) => void,
     ): void;
+
+    /**
+     * WebSocket connection for events.
+     *
+     * @param params.app - Applications to subscribe to.
+     * @param params.subscribeAll - (optional) Subscribe to all Asterisk events. If provided, the applications listed will be subscribed to all events, effectively disabling the application
```
Comment (peterblazejewicz) [inline types/ari-client/index.d.ts:284]:
```suggestion
     * @param [params.subscribeAll] - Subscribe to all Asterisk events. If provided, the applications listed will be subscribed to all events, effectively disabling the application
```

File: types/ari-client/index.d.ts [hunk 6]
```diff
@@ -142,16 +265,48 @@ export type AnyEventType =
...
+     * specific subscriptions. Default is false.
+     */
     eventWebsocket(params: { app: string | string[]; subscribeAll?: boolean }): Promise<Message>;
+
+    /**
+     * Generate a user event.
+     *
+     * @param params.eventName - Event name.
+     * @param params.application - The name of the application that will receive this event.
+     * @param params.source - (optional) URI for event source (channel:{channelId}, bridge:{bridgeId}, endpoint:{tech}/{resource}, deviceState:{deviceName}.
```
Comment (peterblazejewicz) [inline types/ari-client/index.d.ts:294]:
```suggestion
     * @param [params.source] - URI for event source (channel:{channelId}, bridge:{bridgeId}, endpoint:{tech}/{resource}, deviceState:{deviceName}.
```

File: types/ari-client/index.d.ts [hunk 7]
```diff
@@ -142,16 +265,48 @@ export type AnyEventType =
...
+     * @param params.variables - (optional) The "variables" key in the body object holds custom key/value pairs to add to the user event. Ex. { "variables": { "key": "value" } }.
```
Comment (peterblazejewicz) [inline types/ari-client/index.d.ts:295]:
```suggestion
     * @param [params.variables] - The "variables" key in the body object holds custom key/value pairs to add to the user event. Ex. { "variables": { "key": "value" } }.
```

File: types/ari-client/index.d.ts [hunk 8]
```diff
@@ -142,16 +265,48 @@ export type AnyEventType =
...
         params: { eventName: string; application: string; source?: string | string[]; variables?: Containers },
         callback: (err: Error) => void,
     ): void;
+
+    /**
+     * Generate a user event.
+     *
+     * @param params.eventName - Event name.
+     * @param params.application - The name of the application that will receive this event.
+     * @param params.source - (optional) URI for event source (channel:{channelId}, bridge:{bridgeId}, endpoint:{tech}/{resource}, deviceState:{deviceName}.
```
Comment (peterblazejewicz) [inline types/ari-client/index.d.ts:307]:
```suggestion
     * @param [params.source] - URI for event source (channel:{channelId}, bridge:{bridgeId}, endpoint:{tech}/{resource}, deviceState:{deviceName}.
```

File: types/ari-client/index.d.ts [hunk 9]
```diff
@@ -142,16 +265,48 @@ export type AnyEventType =
...
+     * @param params.variables - (optional) The "variables" key in the body object holds custom key/value pairs to add to the user event. Ex. { "variables": { "key": "value" } }.
```
Comment (peterblazejewicz) [inline types/ari-client/index.d.ts:308]:
```suggestion
     * @param [params.variables] - The "variables" key in the body object holds custom key/value pairs to add to the user event. Ex. { "variables": { "key": "value" } }.
```

File: types/ari-client/index.d.ts [hunk 10]
```diff
@@ -160,719 +315,2290 @@ export interface Events {
...
+     * type name of the event to filter The value must be the string name (case sensitive) of the event type that needs filtering. For example: { "allowed": [ { "type": "StasisStart" },
+     * { "type": "StasisEnd" } ] } As this specifies only an allowed list, then only those two event type messages are sent to the application. No other event messages are sent. The following
+     * rules apply: * If the body is empty, both the allowed and disallowed filters are set empty. * If both list types are given then both are set to their respective values (note,
+     * specifying an empty array for a given type sets that type to empty). * If only one list type is given then only that type is set. The other type is not updated. * An empty "allowed"
+     * list means all events are allowed. * An empty "disallowed" list means no events are disallowed. * Disallowed events take precedence over allowed events if the event type
+     * is specified in both lists.
+     *
+     * @param params.applicationName - Applications name.
+     * @param params.filter - (optional) Specify which event types to allow/disallow.
```
Comment (peterblazejewicz) [inline types/ari-client/index.d.ts:1777]:
```suggestion
     * @param [params.filter] - Specify which event types to allow/disallow.
```

File: types/ari-client/index.d.ts [hunk 11]
```diff
@@ -160,719 +315,2290 @@ export interface Events {
...
```
Comment (peterblazejewicz) [inline types/ari-client/index.d.ts:1797]:
```suggestion
     * @param [params.filter] - Specify which event types to allow/disallow.
```

File: types/ari-client/index.d.ts [hunk 12]
```diff
@@ -160,719 +315,2290 @@ export interface Events {
...
+     * type name of the event to filter The value must be the string name (case sensitive) of the event type that needs filtering. For example: { "allowed": [ { "type": "StasisStart" },
+     * { "type": "StasisEnd" } ] } As this specifies only an allowed list, then only those two event type messages are sent to the application. No other event messages are sent. The following
+     * rules apply: * If the body is empty, both the allowed and disallowed filters are set empty. * If both list types are given then both are set to their respective values (note,
+     * specifying an empty array for a given type sets that type to empty). * If only one list type is given then only that type is set. The other type is not updated. * An empty "allowed"
+     * list means all events are allowed. * An empty "disallowed" list means no events are disallowed. * Disallowed events take precedence over allowed events if the event type
+     * is specified in both lists.
+     *
+     * @param params.filter - (optional) Specify which event types to allow/disallow.
```
Comment (peterblazejewicz) [inline types/ari-client/index.d.ts:1937]:
```suggestion
     * @param [params.filter] - Specify which event types to allow/disallow.
```

File: types/ari-client/index.d.ts [hunk 13]
```diff
@@ -160,719 +315,2290 @@ export interface Events {
...
+     */
     getObject(params: { configClass: string; objectType: string; id: string }): Promise<ConfigTuple[]>;
+
+    /**
+     * Create or update a dynamic configuration object.
+     *
+     * @param params.configClass - The configuration class containing dynamic configuration objects.
+     * @param params.objectType - The type of configuration object to create or update.
+     * @param params.id - The unique identifier of the object to create or update.
+     * @param params.fields - (optional) The body object should have a value that is a list of ConfigTuples, which provide the fields to update. Ex. [ { "attribute": "directmedia", "value":
```
Comment (peterblazejewicz) [inline types/ari-client/index.d.ts:1969]:
```suggestion
     * @param [params.fields] - The body object should have a value that is a list of ConfigTuples, which provide the fields to update. Ex. [ { "attribute": "directmedia", "value":
```

File: types/ari-client/index.d.ts [hunk 14]
```diff
@@ -160,719 +315,2290 @@ export interface Events {
...
         callback: (err: Error, configtuples: ConfigTuple[]) => void,
     ): void;
+
+    /**
+     * Create or update a dynamic configuration object.
+     *
+     * @param params.configClass - The configuration class containing dynamic configuration objects.
+     * @param params.objectType - The type of configuration object to create or update.
+     * @param params.id - The unique identifier of the object to create or update.
+     * @param params.fields - (optional) The body object should have a value that is a list of ConfigTuples, which provide the fields to update. Ex. [ { "attribute": "directmedia", "value":
```
Comment (peterblazejewicz) [inline types/ari-client/index.d.ts:1983]:
```suggestion
     * @param [params.fields] - The body object should have a value that is a list of ConfigTuples, which provide the fields to update. Ex. [ { "attribute": "directmedia", "value":
```

File: types/ari-client/index.d.ts [hunk 15]
```diff
@@ -160,719 +315,2290 @@ export interface Events {
...
+     * @param params.configClass - The configuration class containing dynamic configuration objects.
+     * @param params.objectType - The type of configuration object to delete.
+     * @param params.id - The unique identifier of the object to delete.
+     */
     deleteObject(params: { configClass: string; objectType: string; id: string }): Promise<void>;
+
+    /**
+     * Gets Asterisk system information.
+     *
+     * @param params.only - (optional) Filter information returned.
```
Comment (peterblazejewicz) [inline types/ari-client/index.d.ts:2014]:
```suggestion
     * @param [params.only] - Filter information returned.
```

**Review by peterblazejewicz — APPROVED** [review]
LGTM!
Yo could disregard suggestion on `[optional]` JSDoc syntax!

**Comment by dioris-moreno** [comment]
> LGTM!
> Yo could disregard suggestion on `[optional]` JSDoc syntax!

I just committed them. It is much better. Thanks!

**Review by peterblazejewicz — APPROVED** [review]
thx for spending your time on this. JSDocs improves DX when using types,
I'll review remaining part later and submit PR,
thanks!

**Comment by amcasey** [comment]
I didn't dig into the perf failure, but it seems intuitively reasonable that quick info with documentation would be slower than quick info without.

**Pull request merged**